app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -574,3 +583,29 @@ async function signOut(logger: loggerProvider.Logger) {
         await amplify.Auth.signOut()
     }
 }
+
+// ======================
+// === ChangePassword ===
+// ======================
+
+const currentAuthenticatedUser = () =>
+    results.Result.wrapAsync(
+        () => amplify.Auth.currentAuthenticatedUser() as Promise<amplify.CognitoUser>
```

Comment:
`as` is an unsafe cast, do I remember that correctly? If so, can wee do it without `as`?